idaes/generic_models/properties/core/pure/Perrys.py

```diff
@@ -228,33 +228,35 @@ def return_expression(b, cobj, T):
 
 class dens_mol_liq_comp():  # dens_mol_liq_comp_eqn_1, dens_mol_liq_comp_eqn_2
 
+    @staticmethod
     def build_parameters(cobj):
         cobj.dens_mol_liq_comp_coeff_eqn_type = Param(
             mutable=True, doc="Liquid molar density equation form")
 
         try:
             set_param_from_config(cobj, param="dens_mol_liq_comp_coeff",
                                   index="eqn_type")
-            eqn_type = cobj.dens_mol_liq_comp_coeff_eqn_type.value
         except KeyError:
-            _log.info("DEPRECATED - {} dens_mol_liq_comp_coeff index "
-                      "'eqn_type' should be specified, defaulting to "
-                      "equation form 1.".format(cobj))
-            eqn_type = 1  # default to equation form 1 if not specified
+            _log.warning("DEPRECATED - {} dens_mol_liq_comp_coeff index "
+                         "'eqn_type' should be specified, defaulting to "
+                         "equation form 1.".format(cobj))
+            # default to equation form 1 if not specified
+            cobj.dens_mol_liq_comp_coeff_eqn_type.value = 1
 
-        if eqn_type == 1:
+        if cobj.dens_mol_liq_comp_coeff_eqn_type.value == 1:
             dens_mol_liq_comp_eqn_1.build_parameters(cobj)
-        elif eqn_type == 2:
+        elif cobj.dens_mol_liq_comp_coeff_eqn_type.value == 2:
             dens_mol_liq_comp_eqn_2.build_parameters(cobj)
         else:
             raise BurntToast(f"{cobj.name} unrecognized value for "
```

Comment:
Looking at how this is written, this should be a `ConfigurationError`, as we do not have nay checks to make sure that the value of this parameter is in the expected range (sorry, I know I suggested `BurntToast` last time).